Samples/SensorActivityMonitorConsoleApp/SensorActivityMonitorConsoleApp/cameraActivityCallback.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+
+#include "pch.h"
+
+using namespace Microsoft::WRL;
+
+HRESULT MyCameraNotificationCallback::Initialize(_In_z_ LPCWSTR symbolicName)
+{
+    _event = ::CreateEvent(nullptr, FALSE, FALSE, nullptr);
+    RETURN_LAST_ERROR_IF_NULL(_event);
+
+    return StringCchCopy(_symbolicName, MAX_PATH, symbolicName);
+}
+
+IFACEMETHODIMP MyCameraNotificationCallback::OnActivitiesReport(_In_ IMFSensorActivitiesReport* sensorActivitiesReport)
+{
+    bool inUse = false;
+    wil::com_ptr_nothrow<IMFSensorActivityReport> sensorActivity;
+    ULONG totalReportCount = 0;
+
+    // There's two ways to look the activity reports.
+    // option 1. One can ask all the reports and iterate through them, and filter the results if wanted
+    //
+    // option 2. Get only the report for specific device by using the symbolic link name.
+
+    // option 1:
+    printf("\nIterating all activity reports \n");
+
+    RETURN_IF_FAILED(sensorActivitiesReport->GetCount(&totalReportCount));
+    
+    printf("%s@%d: [totalcount=%lu] \n", __FUNCTION__, __LINE__, totalReportCount);
+
+    for (ULONG idx = 0; idx < totalReportCount; idx++)
+    {
+        WCHAR symbolicName[MAX_PATH] = {};
+        ULONG count = 0;
+        wil::com_ptr_nothrow<IMFSensorActivityReport> activityReport;
+        RETURN_IF_FAILED(sensorActivitiesReport->GetActivityReport(idx, &activityReport));
+
+        RETURN_IF_FAILED(activityReport->GetSymbolicLink(symbolicName, MAX_PATH, &count));

```

Comment:
>symbolicName [](http://example.com/codeflow?start=57&length=12)

is this supposed to be compared to the one we are interested in?

Reply:
Yeah I probably should as I'm setting the inUse flag.